daemon/containerd/image_list.go

```diff
@@ -199,6 +199,18 @@ func (i *ImageService) Images(ctx context.Context, opts imagetypes.ListOptions)
 				indexPlatform: img.Target().Platform,
 				dockerImage:   &dockerImage,
 			})
+			target := img.Target()
+
+			var platform ocispec.Platform
+			if target.Platform != nil {
+				platform = *target.Platform
+			} else {
+				platform = dockerImage.Platform
+			}
+
+			if !platformMatcher.Match(platform) {
```

Comment:
Perhaps would be worth adding your commit-message description as a comment here;  this would not be obvious immediately when looking at the code.

> Currently this won't have any real effect because the platform matcher matches all platform and is only used for sorting.


Curious if this _would_ filter, wouldn't that mean that `docker image ls` would not include sizes for all images?

Reply:
I'd assume it would only include the size for matching-platform images. However, I think that makes sense: if the passes in a specific platform matcher, it's because they only want to know about those particular images.

Reply:
Yeah my train of thought here was that;


```
abcd ubuntu:latest 200MB
```


Is the compact view of;

```
abcd ubuntu:latest 200MB
    |- ubuntu:latest X64   100MB
    |- ubuntu:latest ARM64 100MB
```

So when "collapsed", it would show the summary of it all.


For sure, this is a bit of a limitation of the current means we have, and to some extent there's also something to be said for that to be showing "default" platform. But "hiding" the 100MB for the non-matching platform is also surprising (It says I have 200MB used for images, but I only see 100??)

Reply:
Yes, I think it makes more sense to show the total size. Fixed.